android/app/src/main/java/com/zulipmobile/ShareContent.java

```diff
@@ -0,0 +1,93 @@
+package com.zulipmobile;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Allows Sending Content from zulip to other apps.
+ */
+
+public class ShareContent extends ReactContextBaseJavaModule {
+    private String TAG = "ShareContent";
+
+    public ShareContent(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public String getName() {
+        return "ShareContent";
+    }
+
+    @ReactMethod
+    public void shareContent(@Nullable ReadableMap source) {
+        try {
+            String urlString = source.getString("uri");
+            String[] urlSplit = urlString.split("/");
+            String fileName = urlSplit[urlSplit.length - 1];
+            URL url = new URL(urlString);
+            String auth = source.getMap("headers").getString("Authorization");
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.addRequestProperty("Authorization", auth);
+            String mimeType = conn.getContentType();
+            InputStream in = conn.getInputStream();
+
+            // save content in cache directory
+            File contentsFolder = new File(getReactApplicationContext().getCacheDir(), "contents");
+            contentsFolder.mkdir();
+            File file = new File(contentsFolder, fileName);
+            FileOutputStream out = new FileOutputStream(file);
+            IOUtils.copy(in, out);
+            out.flush();
+            out.close();
+            Uri uri = FileProvider.getUriForFile(
+                getReactApplicationContext(),
+                "com.zulipmobile.fileprovider",
+                file
+            );
+
+            // invoke share intent
+            shareContentUri(uri, mimeType);
+        } catch (Exception e) {
```

Comment:
Can we also log this error so that it will get sent to sentry? I'm not sure off the top of my head how to do that in native code, but if you ask in #mobile-team on CZO someone will know.

Reply:
I started a CZO thread for this [here](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/logging.20to.20sentry.20from.20native.20android.20code) :)

Reply:
I've introduced a call to `Sentry.captureException` in the catch block. As far as I could tell by reading the [documentation](https://docs.sentry.io/platforms/android/), this should work, since when we call `Sentry.init` in JavaScript it should also call the corresponding `Sentry.init` in android.

Reply:
I just tested this out by using my own sentry key, and it works fine.